src/v/cluster/controller_backend.cc

```diff
@@ -289,9 +290,69 @@ void controller_backend::setup_metrics() {
       });
 }
 
+static absl::flat_hash_map<model::ntp, model::revision_id>
+create_topic_table_snapshot(
```

Comment:
nit: it will be helpful to have a comment why we make a snpashot and what exactly is in the snapshot.